typescript/sdk/src/token/sealevel/metadata.ts

```diff
@@ -0,0 +1,111 @@
+import { Connection, PublicKey } from '@solana/web3.js';
+import { deserializeUnchecked } from 'borsh';
+
+class Creator {
+  address!: Uint8Array;
+  verified!: number;
+  share!: number;
+
+  constructor(fields: any) {
+    Object.assign(this, fields);
+  }
+}
+
+class MetadataData {
+  name!: string;
+  symbol!: string;
+  uri!: string;
+  sellerFeeBasisPoints!: number;
+  creators?: Creator[];
+
+  constructor(fields: any) {
+    Object.assign(this, fields);
+  }
+}
+
+export class Metadata {
+  key!: number;
+  updateAuthority!: Uint8Array;
+  mint!: Uint8Array;
+  data!: MetadataData;
+  primarySaleHappened!: number;
+  isMutable!: number;
+  editionNonce?: number;
+
+  constructor(fields: any) {
+    Object.assign(this, fields);
+  }
+}
+
+export const SPL_TOKEN_METADATA_SCHEMA = new Map<any, any>([
+  [
+    Creator,
+    {
+      kind: 'struct',
+      fields: [
+        ['address', [32]],
+        ['verified', 'u8'],
+        ['share', 'u8'],
+      ],
+    },
+  ],
+  [
+    MetadataData,
+    {
+      kind: 'struct',
+      fields: [
+        ['name', 'string'],
+        ['symbol', 'string'],
+        ['uri', 'string'],
+        ['sellerFeeBasisPoints', 'u16'],
+        ['creators', { kind: 'option', type: [Creator] }],
+      ],
+    },
+  ],
+  [
+    Metadata,
+    {
+      kind: 'struct',
+      fields: [
+        ['key', 'u8'],
+        ['updateAuthority', [32]],
+        ['mint', [32]],
+        ['data', MetadataData],
+        ['primarySaleHappened', 'u8'],
+        ['isMutable', 'u8'],
+        ['editionNonce', { kind: 'option', type: 'u8' }],
+      ],
+    },
+  ],
+]);
+
+const METADATA_PROGRAM_ID = new PublicKey(
+  'metaqbxxUerdq28cj1RbAWkYQm3ybzjb6a8bt518x1s',
+);
```

Comment:
i don't really understand this

Reply:
you mean, you don't understand why the hardcoded address or the logic for fetching the metadata on chain?

Reply:
yup double checked on the explorer for the SVM chains we support and the program exists on each one